fpl/data/data_converter_class.py

```diff
@@ -0,0 +1,80 @@
+"""Convert data to CSV."""
+import json
+from pathlib import Path
+from tqdm import tqdm
+import pandas as pd
+
+
+class DataConverter:
+    """Data converter class."""
+
+    def __init__(
+        self, data_dir: str, entity: str, main_data_folder="data", main_interim_folder="interim"
+    ):
+        """Set configuration parameters.
+
+        Args:
+            data_dir (str): Path to dir holding JSON dumps of Fantasy Premier League.
+            entity (str): Either "elements" or "teams".
+        """
+        self.data_dir = Path(data_dir)
+        self.entity = entity
+        self.interim_folder_path = Path(main_data_folder, main_interim_folder)
+        self.interim_folder_entity_path = Path(
+            self.interim_folder_path, self.data_dir.name + "_" + self.entity
+        ).with_suffix(".csv")
+
+    def make_interim_folder_if_absent(self):
+        """Generate interim folder if non-existant.
+        """
+        self.interim_folder_entity_path.parent.mkdir(exist_ok=True)
+
+    def _get_game_week(self, data: dict) -> int:
+        """Get the gameweek from an events list.
+
+        Args:
+            data (dict): data dict.
+        Returns:
+            int: Current gameweek
+        """
+        gameweek = list(filter(lambda x: x["is_current"] is True, data["events"]))
+
+        return gameweek[0]["id"] if gameweek else 0
+
+    def convert_json_to_csv_on_entity(self):
+        """Convert multiple JSON into csv according to selected entity.
+
+        Args:
+            data_dir (str): Path to dir holding JSON dumps of Fantasy Premier League.
+            entity (str): Either "elements" or "teams".
+        """
+
```

Comment:
Do we want to validate the value of `entity` to only be "elements" or "teams"?

Reply:
The cli already validates the entity parameter through the ´click.Choice´ method

Reply:
Ok, so we know that we will only be using the method through the CLI? What about when we want to run an entire training pipeline?